diff -rupN src-orig/java1.5/org/jnetpcap/protocol/tcpip/Tcp.java src/java1.5/org/jnetpcap/protocol/tcpip/Tcp.java
--- src-orig/java1.5/org/jnetpcap/protocol/tcpip/Tcp.java	2012-01-31 13:51:53.744516738 +0100
+++ src/java1.5/org/jnetpcap/protocol/tcpip/Tcp.java	2012-01-31 14:10:29.826470061 +0100
@@ -426,47 +426,63 @@ public class Tcp extends JHeaderMap<Tcp>
 	 * @author Sly Technologies, Inc.
 	 */
 	public enum Flag {
+		/** FIN (1 bit) - No more data from sender. */
+		FIN(0),
+		/**
+		 * SYN (1 bit) - Synchronize sequence numbers. Only the first packet sent
+		 * from each end should have this flag set. Some other flags change meaning
+		 * based on this flag, and some are only valid for when it is set, and
+		 * others when it is clear.
+		 */
+		SYN(1),
+		/** RST (1 bit) - Reset the connection. */
+		RST(2),
+		/**
+		 * PSH (1 bit) - Push function. Asks to push the buffered data to the
+		 * receiving application.
+		 */
+		PSH(3),
 		/**
 		 * ACK (1 bit) - indicates that the Acknowledgment field is significant. All
 		 * packets after the initial SYN packet sent by the client should have this
 		 * flag set.
 		 */
-		ACK,
-		/**
-		 * CWR (1 bit) - Congestion Window Reduced (CWR) flag is set by the sending
-		 * host to indicate that it received a TCP segment with the ECE flag set and
-		 * had responded in congestion control mechanism (added to header by RFC
-		 * 3168).
-		 */
-		CWR,
+		ACK(4),
+		/** URG (1 bit) - indicates that the Urgent pointer field is significant. */
+		URG(5), 
 		/**
 		 * ECE (1 bit) - ECN-Echo indicates If the SYN flag is set, that the TCP
 		 * peer is ECN capable. If the SYN flag is clear, that a packet with
 		 * Congestion Experienced flag in IP header set is received during normal
 		 * transmission (added to header by RFC 3168).
 		 */
-		ECE,
-
-		/** FIN (1 bit) - No more data from sender. */
-		FIN,
-		/**
-		 * PSH (1 bit) - Push function. Asks to push the buffered data to the
-		 * receiving application.
-		 */
-		PSH,
-
-		/** RST (1 bit) - Reset the connection. */
-		RST,
+		ECE(6),
 		/**
-		 * SYN (1 bit) - Synchronize sequence numbers. Only the first packet sent
-		 * from each end should have this flag set. Some other flags change meaning
-		 * based on this flag, and some are only valid for when it is set, and
-		 * others when it is clear.
+		 * CWR (1 bit) - Congestion Window Reduced (CWR) flag is set by the sending
+		 * host to indicate that it received a TCP segment with the ECE flag set and
+		 * had responded in congestion control mechanism (added to header by RFC
+		 * 3168).
 		 */
-		SYN,
+		CWR(7),
+    ;
 
-		/** URG (1 bit) - indicates that the Urgent pointer field is significant. */
-		URG, ;
+    /**
+     * Offset of flag in the Integer as returned by Tcp.flags() or passed to 
+     * Flag.asSet().
+     */ 
+    private final int offset;
+
+    Flag(int offset){
+      this.offset = offset;
+    }
+
+    /**
+     * Tests if the flag is set for the given Integer. The flags argument is 
+     * an int as returned by Tcp.flags() or passed to Flag.asSet().
+     */
+    public boolean isSet(int flags){
+      return ((flags & (1 << offset)) > 0);
+    }
 
 		/**
 		 * Converts 8 contigeous bits of an inteteger to a set collection of enum
@@ -482,8 +498,9 @@ public class Tcp extends JHeaderMap<Tcp>
 			final int len = values().length;
 
 			for (int i = 0; i < len; i++) {
-				if ((flags & (1 << i)) > 0) {
-					set.add(values()[i]);
+        Flag f = values()[i];
+				if (f.isSet(flags)) {
+					set.add(f);
 				}
 			}
 
